bool auto_navigation
int32 heading